packages/block-editor/src/utils/transform-styles/index.js

```diff
@@ -2,38 +2,76 @@
  * External dependencies
  */
 import postcss, { CssSyntaxError } from 'postcss';
-import wrap from 'postcss-prefixwrap';
+import prefixSelector from 'postcss-prefix-selector';
 import rebaseUrl from 'postcss-urlrebase';
 
 const cacheByWrapperSelector = new Map();
 
+const ROOT_SELECTORS = [
+	':root :where(body)',
+	':where(body)',
+	':root',
+	'html',
+	'body',
+];
+
+function replaceDoublePrefix( selector, prefix ) {
+	// Avoid prefixing an already prefixed selector.
+	const doublePrefix = `${ prefix } ${ prefix }`;
+	if ( selector.includes( doublePrefix ) ) {
+		return selector.replace( doublePrefix, prefix );
+	}
+	return selector;
+}
+
 function transformStyle(
 	{ css, ignoredSelectors = [], baseURL },
 	wrapperSelector = ''
 ) {
-	// When there is no wrapper selector or base URL, there is no need
+	// When there is no wrapper selector and no base URL, there is no need
 	// to transform the CSS. This is most cases because in the default
 	// iframed editor, no wrapping is needed, and not many styles
 	// provide a base URL.
 	if ( ! wrapperSelector && ! baseURL ) {
 		return css;
 	}
-	const postcssFriendlyCSS = css
-		.replace( /:root :where\(body\)/g, 'body' )
-		.replace( /:where\(body\)/g, 'body' );
 	try {
 		return postcss(
 			[
 				wrapperSelector &&
-					wrap( wrapperSelector, {
-						ignoredSelectors: [
-							...ignoredSelectors,
-							wrapperSelector,
-						],
+					prefixSelector( {
+						prefix: wrapperSelector,
+						exclude: [ ...ignoredSelectors, wrapperSelector ],
+						transform( prefix, selector, prefixedSelector ) {
+							const rootSelector = ROOT_SELECTORS.find(
+								( rootSelectorCandidate ) =>
+									selector.startsWith( rootSelectorCandidate )
+							);
+
+							// Reorganize root selectors such that the root part comes before the prefix,
+							// but the prefix still comes before the remaining part of the selector.
+							if ( rootSelector ) {
+								const selectorWithoutRootPart = selector
+									.replace( rootSelector, '' )
+									.trim();
+								const updatedRootSelector =
+									`${ rootSelector } ${ prefix } ${ selectorWithoutRootPart }`.trim();
+
+								return replaceDoublePrefix(
+									updatedRootSelector,
+									prefix
+								);
+							}
+
+							return replaceDoublePrefix(
+								prefixedSelector,
+								prefix
+							);
```

Comment:
The 'should not double wrap selectors' was failing when I switched to the new library. You're right that the `ignoredSelectors` seems to be where the old library works differently to the new one.

I think this will be an issue as `transformStyles` is a public API, so `ignoredSelectors` should work the same as before. 🤔 

Probably the best bet would be to replicate how the old library works in the callback rather than the way I've done it using `replaceDoublePrefix` 😄

Reply:
Done in [6b43b57](https://github.com/WordPress/gutenberg/pull/64534/commits/6b43b57d5c8c7cba546ba276e95ed5485c0dbcd3)

Reply:
I haven't had a chance to look deeply at the code after the latest updates but did you account for the fact the old plugin allows regex in those ignored selectors?

Here's a snippet from the [plugin's readme](https://github.com/dbtedman/postcss-prefixwrap).
```
    // You may want to exclude some selectors from being prefixed, this is
    // enabled using the `ignoredSelectors` option.
    ignoredSelectors: [":root", "#my-id", /^\.some-(.+)$/],
```

As noted in my earlier comment, this new plugin appears to do a direct comparison of selectors. Take that with a grain of salt but something to double all the same.

Reply:
Yep, the pseudocode is `selector.match( ignoredSelector )`, and `match` which will work for both plain strings and regexes.

Reply:
Nice. Appreciate the extra test coverage there 👌